pennylane/qnn/keras.py

```diff
@@ -270,24 +306,40 @@ def call(self, inputs):
         outputs = []
         for x in inputs:  # iterate over batch
 
-            # The QNode can require some passed arguments to be positional and others to be keyword.
-            # The following loops through input arguments in order and uses functools.partial to
-            # bind the argument to the QNode.
-            qnode = self.qnode
-
-            for arg in self.sig:
-                if arg is not self.input_arg:  # Non-input arguments must always be positional
-                    w = self.qnode_weights[arg]
-                    qnode = functools.partial(qnode, w)
-                else:
-                    if self.input_is_default:  # The input argument can be positional or keyword
-                        qnode = functools.partial(qnode, **{self.input_arg: x})
+            if qml.tape_mode_active():
+                res = self._evaluate_qnode_tape_mode(x)
+                outputs.append(res)
+            else:
+                # The QNode can require some passed arguments to be positional and others to be
+                # keyword. The following loops through input arguments in order and uses
+                # functools.partial to bind the argument to the QNode.
+                qnode = self.qnode
+
+                for arg in self.sig:
+                    if arg is not self.input_arg:  # Non-input arguments must always be positional
+                        w = self.qnode_weights[arg]
+                        qnode = functools.partial(qnode, w)
                     else:
-                        qnode = functools.partial(qnode, x)
-            outputs.append(qnode())
+                        if self.input_is_default:  # The input argument can be positional or keyword
+                            qnode = functools.partial(qnode, **{self.input_arg: x})
+                        else:
+                            qnode = functools.partial(qnode, x)
+                outputs.append(qnode())
 
         return tf.stack(outputs)
 
+    def _evaluate_qnode_tape_mode(self, x):
+        """Evaluates a tape-mode QNode for a single input datapoint.
+
+        Args:
+            x (tensor): the datapoint
+
+        Returns:
+            tensor: output datapoint
+        """
+        kwargs = {**{self.input_arg: x}, **{k: 1.0 * w for k, w in self.qnode_weights.items()}}
```

Comment:
Note the 1.0 here, which converts a `Variable` into a `Tensor`, allowing us to do things like `qml.Rot(*weight)`.

Reply:
sneaky